client/asset/eth/multirpc.go

```diff
@@ -569,12 +569,80 @@ func (m *multiRPCClient) voidUnusedNonce() {
 	m.lastNonce.stamp = time.Time{}
 }
 
-func (m *multiRPCClient) reconfigure(ctx context.Context, settings map[string]string) error {
+// createAndCheckProviders creates and connects to providers. It checks that
+// unknown providers have a sufficient api to trade and saves good providers to
+// file. One bad provider or connect problem will cause this to error.
+func createAndCheckProviders(walletDir string, endpoints []string, net dex.Network,
+	log dex.Logger, ctx context.Context) error {
+	var localCP map[string]bool
+	path := filepath.Join(walletDir, "compliant-providers.json")
+	b, err := os.ReadFile(path)
+	if err == nil {
+		if err := json.Unmarshal(b, &localCP); err != nil {
+			log.Warnf("Couldn't parse compliant providers file: %v", err)
+		}
+	}
+	if localCP == nil {
+		localCP = make(map[string]bool)
+	}
+
+	var writeLocalCP bool
+
+	var unknownEndpoints []string
+	for _, p := range endpoints {
+		d := domain(p)
+		if localCP[d] {
+			continue
+		}
+		writeLocalCP = true
+		localCP[d] = true
+		if _, known := compliantProviders[d]; !known {
+			unknownEndpoints = append(unknownEndpoints, p)
+		}
+	}
+
+	if len(unknownEndpoints) > 0 {
+		ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+		defer cancel()
+
+		providers, err := connectProviders(ctx, unknownEndpoints, log, big.NewInt(chainIDs[net]))
+		if err != nil {
+			return err
+		}
+		defer func() {
+			for _, p := range providers {
+				p.ec.Close()
+			}
+		}()
+		if len(providers) != len(unknownEndpoints) {
+			return errors.New("Could not connect to all unknown providers.")
+		}
+		if err := checkProvidersCompliance(ctx, providers, net, log); err != nil {
+			return err
+		}
+	}
+	if writeLocalCP {
+		// All unknown providers were checked.
+		b, err := json.Marshal(localCP)
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(path, b, 0644); err != nil {
+			log.Errorf("Failed to write compliant providers file: %v", err)
+		}
```

Comment:
Maybe return `error` here instead of just logging it ? So the user can retry saving his configuration (in case it's temporary failure), because he'll have to refill it upon restart ... I can't think of a valid use case to log here instead of erroring right away, thougths ?

Reply:
He won't need to refill it, the providers are saved somewhere else. This file is only for a list of user added providers we didn't know about, like random full nodes. If it fails writing, it would mean the provider needs to be checked again if unknown if doing reconfigure. So, it only saves a little time and not every start up. I can't imagine why this would error here under normal circumstances.

Reply:
I misunderstood it then, thanks for clarifying.